### PR TITLE
More dynamic instrument handler

### PR DESCRIPTION
**Pull request by husobee**
fixes: #291  - attempting to make lookup of the handler function name dynamic from runtime inspection.

### CODE REVIEW

**Comment by husobee** [comment]
Ready for review.

**Comment by husobee** [comment]
Bringing it up to date with master branch.

File: controllers/grants.go [hunk 1]
```diff
@@ -30,15 +33,15 @@ func GrantsRouter(service *grant.Service) chi.Router {
...
-		r.Method("POST", "/", chiware.Throttle(int(throttle))(middleware.InstrumentHandler("RedeemGrants", RedeemGrants(service))))
+		r.Method("POST", "/", chiware.Throttle(int(throttle))(RedeemGrants(service)))
 	} else {
-		r.Method("POST", "/", middleware.InstrumentHandler("RedeemGrants", RedeemGrants(service)))
+		r.Method("POST", "/", RedeemGrants(service))
 	}
 	// Hacky compatibility layer between for legacy grants and new datastore
-	r.Method("GET", "/active", middleware.InstrumentHandler("GetActive", GetActive(service)))
-	r.Method("POST", "/drain", middleware.InstrumentHandler("DrainGrants", DrainGrants(service)))
-	r.Method("POST", "/claim", middleware.InstrumentHandler("ClaimGrant", Claim(service)))
```
Comment (maikelmclauflin) [inline controllers/grants.go:40]: let's match the method names to the names that were previously passed just so we have continuity of data

File: promotion/controllers.go [hunk 2]
```diff
@@ -27,26 +27,30 @@ import (
...
+	r = r.With(middleware.InstrumentHandler)
 	if os.Getenv("ENV") != "local" {
 		r.Method("POST", "/", middleware.SimpleTokenAuthorizedOnly(CreatePromotion(service)))
 	} else {
 		r.Method("POST", "/", CreatePromotion(service))
 	}
 
-	r.Method("GET", "/{claimType}/grants/summary", middleware.InstrumentHandler("GetClaimSummary", GetClaimSummary(service)))
-	r.Method("GET", "/", middleware.InstrumentHandler("GetAvailablePromotions", GetAvailablePromotions(service)))
-	r.Method("POST", "/reportclobberedclaims", middleware.InstrumentHandler("ReportClobberedClaims", PostReportClobberedClaims(service)))
```
Comment (maikelmclauflin) [inline promotion/controllers.go:39]: same here

Comment (husobee) [reply promotion/controllers.go:39]: corrected here: https://github.com/brave-intl/bat-go/pull/296/commits/b2d705881830c8658e40d53260a8f5e9f608308d

File: controllers/grants.go (hunk 1 above)
Comment (husobee) [reply controllers/grants.go:40]: corrected here: https://github.com/brave-intl/bat-go/pull/296/commits/b2d705881830c8658e40d53260a8f5e9f608308d

**Pull request closed without merging**